lbrynet/daemon/Component.py

```diff
@@ -50,26 +51,26 @@ def stop(self):
     def component(self):
         raise NotImplementedError()
 
-    @defer.inlineCallbacks
     def _setup(self):
-        try:
-            result = yield defer.maybeDeferred(self.start)
+        def set_running(result):
             self._running = True
-            defer.returnValue(result)
-        except (defer.CancelledError, AlreadyQuit):
-            pass
-        except Exception as err:
-            log.exception("Error setting up %s", self.component_name or self.__class__.__name__)
-            raise err
-
-    @defer.inlineCallbacks
+            return result
+
+        def handle_error(err):
+            log.error("Error setting up %s\n%s", self.component_name or self.__class__.__name__, err.getTraceback())
+
+        return maybe_deferred_trap_and_trace(
+            (defer.CancelledError, AlreadyQuit), set_running, handle_error
+        )(self.start)()
+
     def _stop(self):
-        try:
-            result = yield defer.maybeDeferred(self.stop)
+        def set_running(result):
             self._running = False
-            defer.returnValue(result)
-        except (defer.CancelledError, AlreadyQuit):
-            pass
-        except Exception as err:
-            log.exception("Error stopping %s", self.__class__.__name__)
-            raise err
+            return result
+
+        def handle_error(err):
+            log.error("Error stopping %s\n%s", self.component_name or self.__class__.__name__, err.getTraceback())
+
+        return maybe_deferred_trap_and_trace(
+            (defer.CancelledError, AlreadyQuit), set_running, handle_error
+        )(self.stop)()
```

Comment:
`)(self.stop)()` looks a bit gnarly :-D

Can `_setup`/`_stop` go back to being `@inlineCallbacks` with the new decorator you've written?